Backpack/Rating/Classes/BPKRatingBubble.m

```diff
@@ -0,0 +1,169 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "BPKRatingBubble.h"
+
+#import <Backpack/Color.h>
+#import <Backpack/Common.h>
+#import <Backpack/Font.h>
+#import <Backpack/Label.h>
+#import <Backpack/Spacing.h>
+
+NS_ASSUME_NONNULL_BEGIN
+@interface BPKRatingBubble ()
+@property(nonatomic) BPKLabel *ratingBubbleLabel;
+@end
+
+@implementation BPKRatingBubble
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+
+    if (self) {
+        [self setUp];
+    }
+
+    return self;
+}
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+
+    if (self) {
+        [self setUp];
+    }
+
+    return self;
+}
+
+- (void)setUp {
+    _lowRatingColor = BPKColor.red500;
+    _mediumRatingColor = BPKColor.yellow500;
+    _highRatingColor = BPKColor.green500;
+
+    self.backgroundColor = BPKColor.gray200;
+
+    self.ratingBubbleLabel = [[BPKLabel alloc] initWithFrame:CGRectZero];
+    self.ratingBubbleLabel.fontStyle = BPKFontStyleTextLgEmphasized;
+    self.ratingBubbleLabel.textColor = BPKColor.white;
+    self.ratingBubbleLabel.lineBreakMode = NSLineBreakByClipping;
+    [self addSubview:self.ratingBubbleLabel];
+
+    [self setUpConstraints];
+    [self updateStyle];
+}
+
+#pragma mark - State setters
+
+- (void)setLowRatingColor:(UIColor *)lowRatingColor {
+    if (_lowRatingColor != lowRatingColor) {
+        _lowRatingColor = lowRatingColor;
+
+        [self updateStyle];
+    }
+}
+
+- (void)setMediumRatingColor:(UIColor *)mediumRatingColor {
+    if (_mediumRatingColor != mediumRatingColor) {
+        _mediumRatingColor = mediumRatingColor;
+
+        [self updateStyle];
+    }
+}
+
+- (void)setHighRatingColor:(UIColor *)highRatingColor {
+    if (_highRatingColor != highRatingColor) {
+        _highRatingColor = highRatingColor;
+
+        [self updateStyle];
+    }
+}
+
+- (void)setRatingValue:(CGFloat)ratingValue {
+    if (_ratingValue != ratingValue) {
+        _ratingValue = ratingValue;
+
+        [self updateStyle];
+    }
+}
+
+#pragma mark - Layout
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+
+    CGFloat radius = CGRectGetHeight(self.bounds) / 2.0f;
+    [self.layer setCornerRadius:radius];
+}
+
+- (void)setUpConstraints {
+    CGFloat ratingBubbleVerticalSpacing = [self ratingBubbleVerticalSpacing];
+
+    self.ratingBubbleLabel.translatesAutoresizingMaskIntoConstraints = NO;
+
+    [NSLayoutConstraint activateConstraints:@[
+        [self.heightAnchor constraintEqualToAnchor:self.ratingBubbleLabel.heightAnchor
+                                          constant:2 * ratingBubbleVerticalSpacing],
+        [self.heightAnchor constraintEqualToAnchor:self.widthAnchor],
```

Comment:
Why do we have to constraints here for the same anchor?

Reply:
Cos i wrote it wrong 😆